web/src/lib/components/pages/record/audio-web.ts

```diff
@@ -111,68 +111,67 @@ export default class AudioWeb {
     this.volumeCallback = cb;
   }
 
-  init() {
-    if (this.initPromise) {
-      return this.initPromise;
+  async init(): Promise<void> {
+    if (this.isInitialized) {
```

Comment:
Where do we set `this.isInitialized` to true?

Reply:
Ugh. Good thing you're paying attention!

I had overlooked that we already have that `ready` flag you mentioned previously, and then additionally failed to correctly implement the lifecycle for this new flag!

I've updated it to use the `ready` flag instead now, whose lifecycle should be fine.